### PR TITLE
Add query directive section in readthedocs

### PR DESCRIPTION
**Pull request by pmantica1**
The vast majority of this pull request is copy-paste. 

I do a couple of controversial things though: 

Remove hyperlinks and mentions of: 
- "Expanding `@optional` vertex fields" . I believe that this section should not be in readthedocs. It is an implementation detail not something that should be documented in readthedocs nor the README.rst. 
- "Performance Penalty", This is specific to OrientDB so we can mention the performance penalty of using `@optional` there.   

I also remove the following paragraph since it is again an implementation detail: 

"If the compiler is able to prove that a type coercion in the @fold` scope is actually a no-op, it may optimize it away. See the `Optional type_equivalence_hints compilation parameter` for more details." 

Also, the `Optional type_equivalence_hints compilation parameter`  section did not contain any more details.

I also mention that the documentation for  `@output_source` section is in development. As explained in this issue https://github.com/kensho-technologies/graphql-compiler/issues/605, the existing documentation for this directive is quite poor.  Also since most of the documentation for `@output_source`, was in a different section, e.g. `Completeness of returned results`, it was a bit hard to include it in this diff.  We can include it's documentation in a later pull request. 

Finally, I make some changes in the exposed schema so that the queries in this page only reference existing GraphQL types.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -57,8 +57,7 @@
 html_static_path = ['_static']
 
 # This specifies how many header levels below the title should we include in the table of contents
-# navigation bar. With a navigation depth of 2 we only include the title and the top header of
-# each page.
+# navigation bar.
 html_theme_options = {
-    'navigation_depth': 2,
+    'navigation_depth': 3,
```
Comment (pmantica1) [inline docs/source/conf.py]: I think the documentation simply looks better with this naviation_depth simply looks better.

File: docs/source/language_specification/schema_types.rst [hunk 2]
```diff
@@ -70,9 +73,9 @@ since we will immediately proceed to dissect the schema.
 
     type Species implements Entity {
         _x_count: Int
+        uuid: ID
         name: String
         alias: [String]
-        description: String
```
Comment (pmantica1) [inline docs/source/language_specification/schema_types.rst]: This field is not used anywhere.

**Comment by pmantica1** [comment]
Here is what the documentation looks like:
https://graphql-compiler.readthedocs.io/en/latest/language_specification/query_directives.html

**Comment by bojanserafimov** [comment]
"Expanding @optional vertex fields" belongs in the readme because the user should know why queries are slow, and know that nested optionals are trouble. I agree that it doesn't belong here, but in a section describing OrientDB-specific exceptions.

**Comment by bojanserafimov** [comment]
"Expanding @optional vertex fields" belongs in the "Performance Penalty" section. The user should know where to expect performance penalties, and why they're necessary. Might also be a good idea to call it "OrientDB Performance Penalties".

Do you plan to do this in this diff, or the one immediately following this one?

**Comment by pmantica1** [comment]
> "Expanding @optional vertex fields" belongs in the "Performance Penalty" section. The user should know where to expect performance penalties, and why they're necessary. Might also be a good idea to call it "OrientDB Performance Penalties".
> 
> Do you plan to do this in this diff, or the one immediately following this one?

The one immediately following one.

**Pull request merged**